crowdflower_results_into_training_data.py

```diff
@@ -7,138 +7,167 @@
-                            for iob_tag in iob_tagged:
-                                if iob_tag[0].decode('utf-8') == lines[i][2]:
-                                    lines[i].pop()
-                                    lines[i].append(iob_tag[1])
-            for l in lines:
-                # Skip <strong> tags
-                if '<strong>' not in l and '</strong>' not in l:
-                    print l
-                    output.append('\t'.join(l) + '\n')
-    with codecs.open(output_file, 'wb', 'utf-8') as o:
-        o.writelines(output)
-    return 0
+                else:
+                    lines[i].append('O') # IOB-tag
+
+                # find part of entity associated with the processed line 
+                for entity, tokens in annotations['entities'].iteritems():
+                    for token, tag in tokens:
+                        if token.decode('utf-8') == lines[i][2]:
+                            lines[i][-1] = tag
+
+        output.extend(lines)
+ 
+    return output
+
+
+def main(crowdflower_csv, pos_data_dir, output_file, debug):
+    results = read_full_results(crowdflower_csv)
+    if debug:
+        print 'Results from crowdflower'
+        print json.dumps(results, indent=2)
+
+    set_majority_vote_answer(results)
+    if debug:
+        print 'Computed majority vote'
+        print json.dumps(results, indent=2)
+
+    tag_entities(results)
+    if debug:
+        print 'Entities tagged'
+        print json.dumps(results, indent=2)
+
+
+    output = produce_training_data(results, pos_data_dir)
+    if debug:
+        print 'Final Output'
+        print '\n'.join(repr(l) for l in output)
+
+
+    output_file.writelines('\t'.join(l).encode('utf-8') + '\n'
+                           for l in output
+                           if '<strong>' not in l and '</strong>' not in l)
 
 if __name__ == "__main__":
-    annotazione_test = json.loads("""
-    { "40": {
-        "": {
-        "answers": []
-        },
-        "frame": "Bio_Parco",
-        "lu" : "pubblicare",
-        "Pubblicatore": {
-        "majority": "Fedele anglicano",
-        "judgments": 3,
-        "answers": [
-            "tizio",
-            "caio",
-            "sempronio"
-        ]
-        },
-        "Opera": {
-        "majority": "opere a difesa dell' anglicanesimo",
-        "judgments": 3,
-        "answers": [
-        "tizio",
-        "caio",
-        "sempronio"
-        ]
-        }
-        }
-    }
-    """
-    )
-    if len(sys.argv) == 4:
-        results = read_full_results(sys.argv[1])
-        #print json.dumps(results, indent=2)
-        maj = set_majority_vote_answer(results)
-        #print json.dumps(maj, indent=2)
-        produce_training_data(maj, sys.argv[2], sys.argv[3])
-    else:
-        print "Usage: %s <CROWDFLOWER_FULL_RESULTS_CSV> <POS_DATA_DIR> <OUTPUT_FILE>" % __file__
-        sys.exit(1)
+    parser = argparse.ArgumentParser()
```

Comment:
Good work!
Could you please wrap the creation of the cli parser into a `create_cli_parser()` function?